From 876ea6fec971df3f81e527bf69648d89f50edc4a Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Fri, 17 May 2024 17:21:06 +0800
Subject: [PATCH 097/137] mtk: hostapd: get link channel information and
 synchronize to AP

'wpa_s->valid_links' is used to determine the connection is MLO or not,
and different ways are used to retrieve operating channel information.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 wpa_supplicant/ucode.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/wpa_supplicant/ucode.c b/wpa_supplicant/ucode.c
index d4b2160b6..450780737 100644
--- a/wpa_supplicant/ucode.c
+++ b/wpa_supplicant/ucode.c
@@ -309,10 +309,16 @@ uc_wpas_iface_status(uc_vm_t *vm, size_t nargs)
 		else if (is_6ghz_freq(bss->freq))
 			band_idx = 2;
 
-		wpa_drv_channel_info(wpa_s, &ci);
-		freq = ci.frequency;
-		center_freq1 = ci.center_frq1;
-		chwidth=ci.chanwidth;
+		if (wpa_s->valid_links) {
+			freq = wpa_s->links[link_id].freq;
+			center_freq1 = wpa_s->links[link_id].center_freq1;
+			chwidth = wpa_s->links[link_id].width;
+		} else {
+			wpa_drv_channel_info(wpa_s, &ci);
+			freq = ci.frequency;
+			center_freq1 = ci.center_frq1;
+			chwidth=ci.chanwidth;
+		}
 
 		if (bss->freq != center_freq1) {
 			if (is_24ghz)
-- 
2.18.0

